lib/linguist/strategy/extension.rb

```diff
@@ -15,9 +20,15 @@ class Extension
       # Selected languages must be in the candidate list, except if it's empty,
       # in which case any language is a valid candidate.
       def self.call(blob, candidates)
+        return candidates if generic? blob.name
```

Comment:
We probably want to ensure these are strings, especially if you plan to add numbered extensions in future:

```suggestion
        return candidates if generic? blob.name.to_s
```